modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/Conditions.es.js

```diff
@@ -130,8 +152,14 @@ function FieldLeft({fields, left, onChange}) {
 					},
 				]}
 				onChange={onChange}
+				onSelectionChange={(itemKey) => {
+					setFieldLeftSelectedKey(itemKey);
+					setOperatorValue(undefined);
```

Comment:
Hi @rebsilva. Me and Daniel noticed a limitation on ClayPicker when we use `null` so we avoided using it because it breaks the keyboard selection. We don't know if it intended or it's a bug.

[claypicker Bug using null.webm](https://github.com/liferay-forms/liferay-portal/assets/26491796/68c6d3ae-0b2c-443e-80d7-b8b2fc422900)

Reply:
oh i see! so its better indeed to keep the `undefined`, we can investigate when we have the time why the Picker doesn't work with null. Thanks a lot Alininha! :cherry_blossom:

Reply:
but i think this behavior is expected indeed, because this `selectedKey` is exactly to say which option is to be selected, if it is `null` or empty string like `' '` i won't select anything right? that's why some bugs are happening related to this on other places...but i don't know why with `undefined` it works, maybe since it is `undefined` they set on the first option by default? maybe this is why it happens

Reply:
since this PR is critical we can leave this way for now, but we can analyse when we have time if we can change this approach to avoid using `undefined` in this case